src/main/java/core/basesyntax/report/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.report;
+
+import core.basesyntax.service.ShopService;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private ShopService shopService;
+
+    public void setShopService(ShopService shopService) {

```

Comment:
This method is not required if we pass `ShopService` via constructor. It's a good practice to make your classes immutable if possible. This is a code quality issue.